index.js

```diff
@@ -130,6 +130,8 @@ class CMSClient {
 
       await this._fetchResourceMetadata();
 
+      if (!this.fetchOptions.includeMetadata) delete this.fetched.metadata;
```

Comment:
Great work per usual, @hawyar . A couple style / design questions for you / @shanejearley :

1. Do we use switch statements as a company? They have drawbacks and pushback from the broader JavaScript community, but this here is a simple use case and it appears to work just fine here. Not pushing one way or the other, just throwing this out there because it came to mind when I saw it.
2. What's the reason for using the delete operator instead of setting value to a falsy value in line 133 (and other places) `delete this.fetched.metadata`? To reduce space? Arbitrary design choice? Or perhaps another valid reason.

Again, code appears functional and well-written. Just bringing this up for discussion to learn and improve together. :)

Reply:
Commented on the switch statement on line 136, Chris. This might just be one of those cases where the switch works well!

Reply:
@ccali11 most of these design decisions were based on my early assumptions with some arbitrary design choices, they are all subject to change.

1. We can remove the switch statement and have json as the default output and only change if client chooses csv. We only have two output types so I don't see the reason for the switch statement anymore.

2. You are correct, using `delete` is less performant. I could have just set it to `undefined` however I did want to return undefined properties to the client that might cause trouble when they iterate over the returned object. What about creating a new object instead of mutating?

Reply:
Sounds good on all fronts!